### ISSUE TITLE
udisksctl does not work without attached tty

### ISSUE THREAD

**Issue by norbusan** [storaged-project/udisks#667]
We are trying to auto-mount a device's partition and carry out some file detection after mounting. We realized this by adding an udev action that does nothing else then
```
sudo -u <USER> /path/to/program $DEVNAME
```
(`DEVNAME` is set by the udev rule).

Within `program` we want to mount some/all of the partitions of the device using `udisksctl`. Since we are running Debian stretch, we have added a localauthority file that allows the user to do this.

Running the above sudo invocation as root manually works without any problems, but triggering it from udev breaks when executing the `udisksctl` call.

It seems that without an attached tty, even when adding `--no-user-interaction`, `udisksctl` does not work.

Is this a known issue, and do you have any suggestion to circumvent that?

Thanks

**Comment by tbzatek** [comment]
Yeah, that may sound like a bug. Does is print an error or any other message?

Anyway, please beware that manipulating block devices within udev rules is known to be problematic and you may run into various issues. When a udev rule is run, the block device is still in a probing phase and udev may hold exclusive lock, denying access to other processes. Also, on some systems this may run in a different namespace, leading to the mounts not being visible to the rest of the system, etc.

Exactly this case - automounting from within udev rules - is known to have issues leading from an inappropriate use as reported by other users.

The proper way would be to write a small udisks client that watches the D-Bus object manager for new block device appearance and acting afterwards (i.e. automounting).

**Comment by norbusan** [comment]
Hi @tbzatek 
thanks for your comments. No, there is no error output as far as I see. I am aware of the mounting problem, and thus I have mitigated the current problem by completely switching away from udisksctl to usbmount, which triggers systemd units to automount (the recommended way). I am also aware of the private namespace, but even with working around that (udev setting) udisksctl didn't react as expected.

Anyway, from my side there is no need to investigate this issue anymore, since I switched to usbmount and do not use udiskctl anymore. I leave the decision up to you.

Ah, yes, d-bus .. I am not sure this one is permanently running on a raspbian device, though...

Best

Norbert
